core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1594,11 +1591,12 @@ private void deleteInternal(LockedInodePath inodePath, boolean replayed, long op
         tempInodePath.setDescendant(delInode, delInodePair.getFirst());
         // Do not journal entries covered recursively for performance
         if (delInode.getId() == inode.getId() || unsafeInodes.contains(delInode.getParentId())) {
-          mInodeTree.deleteInode(tempInodePath, opTimeMs, deleteOptions, journalContext,
-              blockDeletionContext);
+          mInodeTree.deleteInode(rpcContext, tempInodePath, opTimeMs, deleteOptions);
+
         } else {
-          mInodeTree.deleteInode(tempInodePath, opTimeMs, deleteOptions,
-              NoopJournalContext.INSTANCE, blockDeletionContext);
+          mInodeTree.deleteInode(
+              new RpcContext(rpcContext.getBlockDeletionContext(), NoopJournalContext.INSTANCE),
```

Comment:
why a new rpc context? can we reuse `rpcContext`?

Reply:
we can't reuse `rpcContext` because we want to temporarily replace the journal context with `NoopJournalContext.INSTANCE`. I want to keep `RpcContext` immutable so that it's guaranteed that the contexts it is initialized with will be closed when the `RpcContext` closes.

Reply:
ack